### ISSUE TITLE
Class subscription with `__class_getitem__` not working

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#1413]
## Environment data

-   Language Server version: 2021.6.1-pre.1
-   OS and version: Windows 7
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.10 Anaconda

## Sample

```python
class C:
    def __class_getitem__(cls, key) -> str:
        return "C"

s = C[0]
```

## Expected behavior

According to the signature of `C.__class_getitem__`, type of `C[...]` should be `str`.

## Actual behavior

![image](https://user-images.githubusercontent.com/15389744/121284587-af705880-c90f-11eb-8412-6724966fded5.png)

**Comment by erictraut** [comment]
`__class_getitem__` is meant to "Return an object representing the specialization of a generic class by type arguments found in key" according to [the official Python documentation](https://docs.python.org/3/reference/datamodel.html#emulating-generic-types).

Can you explain what you're trying to do here?

**Comment by Azureblade3808** [comment]
I was writing something that supposed to be used like constants but with some arguments.

For example, there is a member in my module named `RSV` (Raw Stochastic Value) and it would be used with the form like `RSV[10]` ( a 10-day RSV factor ). The square brackets are here to make the expression differ from the form of function calls in visual. `RSV[10]` itself would be considered a constant, while `RSV[5]` would be considered another constant. I think it would be much more intuitive that `RSV[10] is RSV[10]` than `RSV(10) is RSV(10)`.

I used to implement RSV as an instance of a private factory class, like
```python
class _RSV:
    def __getitem__(self, key): ...

RSV = _RSV()
```
but then I found that there is the `__class_getitem__` method and tried to eliminate one redundant class.

So, if `__class_getitem__` is not supposed to be used like this, I will revert my recent changes.
